homeassistant/components/sensor/linky.py

```diff
@@ -0,0 +1,86 @@
+"""
+Support for Linky.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.linky/
+"""
+import logging
+import json
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_TIMEOUT
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pylinky==0.1.6']
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=10)
+DEFAULT_TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Configure the platform and add the Linky sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    timeout = config[CONF_TIMEOUT]
+
+    from pylinky.client import LinkyClient
+    client = LinkyClient(username, password, None, timeout)
```

Comment:
If the credentials are wrong make the platform setup fail.

Reply:
Did you mean "valid" data?
Otherwise, there is no public login function. Everything is done in the `fetch_data` function.

Reply:
You need to make the first request to the API during the platform setup and go for the [`PyLinkyError`](https://github.com/Pirionfr/pyLinky/blob/master/pylinky/client.py#L54).